app/models/league.rb

```diff
@@ -0,0 +1,9 @@
+class League < ApplicationRecord
+  belongs_to :user
+  belongs_to :competition
+  has_many :memberships, dependent: :destroy
+  # TODO: Do we create a membership for the owner? If not, we need another method for all
+  has_many :users, through: :memberships
+  validates :name, presence: true
+  validates :password, presence: true
```

Comment:
Should we allow people to create public leagues without a password? In case you don't have a group of friends into soccer to compare yourself to?

Reply:
I had "public" leagues as priority number 2 in the user stories. but you're right that password is still an extra step. i based it on a previous fantasy one i've used. maybe just a join link would suffice?

Reply:
Ah I see, I get it. You meant that the password is like the one on Among Us where you join a specific league with the code? Let's think about joining mechanism indeed, and we'll see when to introduce the password, I think it can be done without initially (public leagues would seem easier than private ones... same thing without restricted permissions, no?)

Reply:
yes exactly. public ones are easier. private ones are the reason why is started the last app 😆